x/interchainstaking/keeper/grpc_query.go

```diff
@@ -146,6 +146,26 @@ func (k *Keeper) DelegatorIntent(c context.Context, req *types.QueryDelegatorInt
 	return &types.QueryDelegatorIntentResponse{Intent: &intent}, nil
 }
 
+// DelegatorIntents returns information about the delegation intent of the caller for all zones.
+func (k *Keeper) DelegatorIntents(c context.Context, req *types.QueryDelegatorIntentsRequest) (*types.QueryDelegatorIntentsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	intents := []*types.DelegatorIntentsResponse{}
+
+	k.IterateZones(ctx, func(_ int64, zone *types.Zone) bool {
+		intent, _ := k.GetDelegatorIntent(ctx, zone, req.DelegatorAddress, false)
+
+		intents = append(intents, &types.DelegatorIntentsResponse{ChainId: zone.ChainId, Intent: &intent})
+		return false
```

Comment:
I think we should not append in the cases where `intent` is empty.  Here, we are appending for all zones, whether or not there is any info.

Reply:
added a nil check to ensure we don't append empty intents